lib/alloc_trace.cpp

```diff
@@ -64,9 +64,6 @@ void AllocTrace::count(void **bt, size_t sz) {
 }
 
 std::ostream &operator<<(std::ostream &out, const AllocTrace &at) {
-#if HAVE_LIBGC
-    PauseTrace temp_pause;
-#endif
```

Comment:
Why is this removed? I don't know what it does, but it does not seem very related.

Reply:
A warning popped out https://github.com/p4lang/p4c/actions/runs/9003590407/job/24734561146. Not sure what triggered it to appear only now but this piece of code does seem unused.

Reply:
Weird. I would expect it to be a guard that does something in its constructor and destructor (based on the name).

Reply:
I can alternatively just add a NOLINT.

Reply:
Indeed, see https://github.com/p4lang/p4c/blob/main/lib/alloc_trace.h#L67. I don't think this should be removed. Is it possible this build (which produced the warning) is without GC? I think the "right" solution to suppress the warning might be to put and explicit destructor there even for the case where the `HAVE_LIBGC` is false.

Reply:
`alloc_trace.h` did not include config and removing `config.h` from `big_int_util.h` caused the `#define` to be missing.